src/config.js

```diff
@@ -33,11 +33,12 @@ function createGlobalConfig () {
     // The native layer will inject a randomised secret here and use it to verify the origin
     let secret = 'PLACEHOLDER_SECRET'
 
-    let isDDGApp = /(iPhone|iPad|Android|Mac).*DuckDuckGo\/[0-9]/i.test(window.navigator.userAgent) ||
-        isApp ||
-        isTopFrame
-    const isAndroid = isDDGApp && /Android/i.test(window.navigator.userAgent)
-    const isMobileApp = isDDGApp && !isApp
+    // @ts-ignore
+    const isDDGApp = ['ios', 'android', 'macos', 'windows'].includes(userPreferences?.platform.name)
+    // @ts-ignore
+    const isAndroid = userPreferences?.platform.name === 'android'
+    // @ts-ignore
+    const isMobileApp = ['ios', 'android'].includes(userPreferences?.platform.name)
```

Comment:
Here I'm reading from the `userPreferences` instead of the UA which we change sometimes remotely, thus breaking our flow (was happening on Google). AFAICT all platforms now use string replacement so this stuff is always available, but if Windows uses a different strategy, this may break.